### PR TITLE
Support `PartialMerkleTree` to be used as secret input in the `.input` file.

### PR DESCRIPTION
**Pull request by Overcastan**
This small PR makes it possible to use `PartialMerkleTree` in the `.input` files as a secret input.

### CODE REVIEW

File: docs/src/intro/usage.md
Comment (bobbinth) [inline docs/src/intro/usage.md:82]: Should we say: "The internal tuple represents the leaf depth and index at this depth, and the hex string represents the leaf value (4 elements)."?

File: miden/src/cli/data.rs
Comment (bobbinth) [inline miden/src/cli/data.rs:55]: Same comment as above re potentially changing "node" -> "leaf".

File: miden/src/cli/data.rs
Comment (bobbinth) [inline miden/src/cli/data.rs:513]: minor nit: could we add our usual section separator above to indicate that this is a start of the "TEST" section?

File: miden/src/cli/data.rs
Comment (bobbinth) [inline miden/src/cli/data.rs:589]: Now looking at this, I think we should use "0x" prefix fox hex values to make it consistent with how we use the `push.0x...` instruction.

**Review by bobbinth — COMMENTED** [review]
Looks great! Thank you! Not a full review yet, but I did leave a couple of nits inline.

One other comment: I think it would be good to put together an example to illustrate how different input sections can be used. This example would go [here](https://github.com/0xPolygonMiden/miden-vm/tree/next/miden/examples) and it the input file could be something similar to what you've put together for tests in this PR.

File: miden/examples/fib_with_merkle_store/fib.masm
Comment (bobbinth) [inline miden/examples/fib_with_merkle_store/fib.masm:6]: Would be good if the example illustrates how we can read from various Merkle tree structures which which we instantiated the advice provider (e.g., using `mtree_read` instruction).

File: miden/src/cli/data.rs
Comment (bobbinth) [inline miden/src/cli/data.rs:213]: One other thing that could be interesting here (and for the other cases above) is to print out the root of the Merkle structure added to the Merkle store. For example:
```
Added Partial Merkle tree with root {root} to the Merkle store
```
This will help user figure out which root can be used in their program.

File: miden/examples/merkle_store/merkle_store.masm
Comment (bobbinth) [inline miden/examples/merkle_store/merkle_store.masm:7]: For this and other cases below, I'd do the following:

- Update the comment to say something like: "get value at depth=2, index=0 from the Partial Merkle Tree; this value should be 20"
- Add assertions after `mtree_get` to verify that the value retrieved was indeed 20.

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you! I left one nit inline. Once it is addressed, we can merge.

File: miden/examples/merkle_store/merkle_store.masm
Comment (bobbinth) [inline miden/examples/merkle_store/merkle_store.masm:31]: I think there is a typo here: shouldn't index also be `2`?

Comment (Overcastan) [reply miden/examples/merkle_store/merkle_store.masm:31]: Indeed. Sorry, I missed it

**Comment by Overcastan** [comment]
Let's merge this PR after the [change parse hex strings](https://github.com/0xPolygonMiden/miden-vm/pull/1076) one, I want to be sure that this example is working.

**Comment by bobbinth** [comment]
We could also merge now, and just make sure these tests work in the `push` PR.

**Pull request merged**
